tests/TestHarness/transactions.py

```diff
@@ -152,48 +152,60 @@ def transferFundsAsync(self, source, destination, amountStr, memo="memo", force=
         return popen, cmdArr
 
     # publish contract and return transaction as json object
-    def publishContract(self, account, contractDir, wasmFile, abiFile, waitForTransBlock=False, shouldFail=False, sign=False):
+    def publishContract(self, account, contractDir, wasmFile, abiFile, waitForTransBlock=True, shouldFail=False, sign=False, retryNum:int=5):
```

Comment:
Changing the default `waitForTransBlock` value here could have repercussions in the tests where it was assumed that the default is `False` and could be relying on that.  I might recommend leaving the default as `False` but setting it to `True` in your test case you're trying to fix.  That way you don't have to audit every instance of the tests that use `publishContract`.

Reply:
I did audit all calls. I thought about removing the option actually because it should always be `True`, but decided the bigger diff was not worth it. There is only one case were we were not passing `True` and it should be true.